Sources/NIO/SocketChannel.swift

```diff
@@ -436,6 +438,16 @@ final class ServerSocketChannel: BaseSocketChannel<ServerSocket> {
             // These are errors we may be able to recover from. The user may just want to stop accepting connections for example
             // or provide some other means of back-pressure. This could be achieved by a custom ChannelDuplexHandler.
             return false
+        case EINVAL:
+            if case .function(let function) = err.reason {
+                // see https://github.com/apple/swift-nio/issues/1030
+                // on Darwin, fcntl(fd, F_SETFL, O_NONBLOCK) sometimes returns EINVAL...
+                //
+                // the comparison is so weird because StaticString isn't Equatable and doesn't have an == method...
+                return String(describing: function) != "fcntl(descriptor:command:value:)"
```

Comment:
Yuck, this seems like a really easy way to break the code later on. Can we push this error handling deeper into the code in any way? Potentially by catching the fcntl EINVAL and either rethrowing or doing something else? I kinda hate this string-based coupling.

Reply:
In fact, maybe `accept` should swallow this.

Reply:
@Lukasa I think we should deliver it

Reply:
In that case, `accept` should wrap this error into something clearer.

Reply:
@Lukasa agreed, implemented that yesterday but github still shows the 'outdated' change :|